primitives/npos-elections/compact/src/lib.rs

```diff
@@ -197,17 +197,16 @@ fn struct_def(
 			fn unique_targets(&self) -> Vec<Self::Target> {
 				// NOTE: this implementation returns the targets sorted, but we don't use it yet per
 				// se, nor is the API enforcing it.
-				let mut all_targets: Vec<Self::Target> = Vec::with_capacity(self.average_edge_count());
+				use sp_std::collections::btree_set::BTreeSet;
```

Comment:
mind moving this up top?

Reply:
I don't think we really can: this code is emitted by the macro, so if we have the `use` statement at the top of this file, I don't think `rustc` will have access to it within the expansion. There isn't AFAIK a good way to put this usage outside the generated scope.

Reply:
ahh then we use should bind this to `$crate`. Now, if some crate does not have `sp_std` in scope or rename it to something else, this will not work.

Reply:
See `fn import()`, and I believe following the pattern there should be easy enough: re-export `sp_std` with doc(hidden) and use it here. 

Actually the current code is also bad. I am using `Vec` assuming that it is always in-scope, while I presume in some context it might not be. Ideal is to use `_npos::sp_std::prelude::Vec` everywhere instead of it. Would be quite ugly but I think it is the correct way to go.

Reply:
I think de7c7b296 should take care of this.

Reply:
Indeed, thanks! 

There are a few more instance of `Vec` in other files of the crate as well. Mind taking care of those as well for consistency?

Reply:
No problem.

Reply:
https://github.com/paritytech/substrate/pull/8314/commits/8788c4b3a52ebda8f4a7e456de28d6061d2ed30c should be the last of them. There is one stdlib `Vec` remaining, and that one is in the macro itself, not in the generated code.